wfsim/core/pulse.py

```diff
@@ -199,6 +210,41 @@ def uniform_to_pe_arr(self, p, channel=0):
         indices = (p * 2000).astype(np.int64) + 1
         return self.__uniform_to_pe_arr[channel, indices]
 
+    def add_truth(self,
+                  photon_timings,
+                  photon_gains,
+                  dt,
+                  channel,
+                  n_double_pe,
+                 ):
+
+        if channel in self.config['turned_off_pmts']:
+            return
+
+        if str(channel) in self.config.get('special_thresholds', {}):
+            threshold = self.config['special_thresholds'][str(channel)] - 0.5
+        else:
+            threshold = self.config['zle_threshold'] - 0.5
+
+        reminder = (photon_timings % dt).astype(int)
+        above_threshold = photon_gains * self.current_max[reminder] * self.current_2_adc > threshold
```

Comment:
Sorry I might be a bit slow, but I must admit that I don't fully grasp what you do here, could you help me a bit?

Reply:
Yeah, it's quite confusing.
  - Current max is the highest value in the SPE pulse model given a reminder [0-9] ns (offset from 10 ns sample time). The SPE pulse model sums up to 0.1 (pe/ns), and the peak is ~0.03 (pe/ns) 
  - Multiply this by the sampled gain of each photon, we get (electron/ns)
  - The current_2_adc take into account n_electron -> current -> voltage -> amplification -> digitization, converting the electron/ns to adc value.

Reply:
Thanks a lot, this is instructive indeed 👍

Reply:
I'll add it to the code as a comment if you don't mind? 

I think there is also a small typo, it should be `remainder` right?